### PR TITLE
CALCITE-3155 : Fixing bug in the unparse of UNION_ALL SqlSetOperator when it has no operands

### PR DESCRIPTION
**Pull request by MusbahELFIL**
when we have a query of the form "select X from TEST_TABLE where Y= ( select Z from TEST-TABLE2 where 1=2), when converting to a RelNode, the plan contains a LogicalValues with no tuples (which is correct since the right hand side should return no tuples).

When converting the above RelNode to an SqlNode the LogicalValues is being converted to a UNION_ALL operator with no operands. Its toString() method which is called when doing an unparse (when converting the generated SqlNode back to an SQL string) is throwing an exception (since it has no tuples).

The fix converts the values with no tuples to a "select NULL, NULL ... from DUAL"

### CODE REVIEW

**Comment by MusbahELFIL** [comment]
Corresponding JIRA issue: https://issues.apache.org/jira/browse/CALCITE-3155

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 1]
```diff
@@ -490,10 +505,10 @@ public boolean hasTrickyRollup(Sort e, Aggregate aggregate) {
     return !dialect.supportsAggregateFunction(SqlKind.ROLLUP)
         && dialect.supportsGroupByWithRollup()
         && (aggregate.getGroupType() == Aggregate.Group.ROLLUP
-            || aggregate.getGroupType() == Aggregate.Group.CUBE
-                && aggregate.getGroupSet().cardinality() == 1)
+        || aggregate.getGroupType() == Aggregate.Group.CUBE
+        && aggregate.getGroupSet().cardinality() == 1)
         && e.collation.getFieldCollations().stream().allMatch(fc ->
-            fc.getFieldIndex() < aggregate.getGroupSet().cardinality());
+        fc.getFieldIndex() < aggregate.getGroupSet().cardinality());
```
Comment (hsyuan) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Can you revert these indent changes?

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 2]
```diff
@@ -390,7 +390,22 @@ public Result visit(Values e) {
                 new SqlIdentifier("DUAL", POS), null, null,
                 null, null, null, null, null));
       }
-      if (list.size() == 1) {
+      if (list.isEmpty()) {
+        //In this case we need to construct the following query:
+        // SELECT NULL as C0, NULL as C1, NULL as C2 ... FROM DUAL WHERE FALSE
+        //This would return an empty result set with the same number of columns as the field names.
```
Comment (hsyuan) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Please leave a space after //

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java (hunk 1 above)
Comment (MusbahELFIL) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Done in the latest commit.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java (hunk 2 above)
Comment (MusbahELFIL) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Done in the latest commit.

**Pull request closed without merging**